### PR TITLE
real part of M returned if values are real

### LINKED ISSUE #14330: Matrix exponential of real number has complex numbers
All the values of matrix `A` are real. But, matrix exponential of `A` has complex numbers. Is this used somewhere?
```
>>> A = Matrix([[1,-1],[1,1]])
>>> exp(A)
Matrix([
[     exp(1 - I)/2 + exp(1 + I)/2, -I*exp(1 - I)/2 + I*exp(1 + I)/2],
[-I*exp(1 + I)/2 + I*exp(1 - I)/2,      exp(1 - I)/2 + exp(1 + I)/2]])
```

### PR DESCRIPTION
**Pull request by vishalg2235**
If values of a matrix are real then matrix exponential should also have real values.

Fixes #14330 
```
>>> a = Matrix([[1,-1],[1,1]])
>>> a.exp()
Matrix([
[E*cos(1), -E*sin(1)],
[E*sin(1),  E*cos(1)]])
```
Ping @normalhuman @jksuom

### CODE REVIEW

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -2517,10 +2517,14 @@ def _jblock_exponential(b):
 
         blocks = list(map(_jblock_exponential, cells))
         from sympy.matrices import diag
+        from sympy import re
         eJ = diag(*blocks)
         # n = self.rows
         ret = P * eJ * P.inv()
-        return type(self)(ret)
+        if all(value.is_real for value in self.values()):
+            return re(ret)
```
Comment (normalhuman) [inline sympy/matrices/matrices.py]: This should be `type(self)(re(ret))`. Some matrices are mutable, some are immutable. The reason for `type(self)` is that exp(M) should have the same type as M.   As written, this code will return an immutable matrix even if self was mutable.

**Comment by normalhuman** [comment]
Apart from matrix type issue, this looks good and should help the matrix exponential be more usable when solving ODE systems. Current behavior: 
```
>>> t = symbols('t', real=True)
>>> exp(t*Matrix([[1, -1], [1, 1]]))
Matrix([
[-I*t*exp(t*(1 + I))/(2*(t - t*(1 + I))) + I*t*exp(t*(1 - I))/(2*(t - t*(1 - I))), I*t**2*exp(t*(1 + I))/(2*(t - t*(1 - I))*(t - t*(1 + I))) - I*t*(t/(t - t*(1 - I)) + 2*I)*exp(t*(1 - I))/(2*(t - t*(1 - I)))],
[                                         I*exp(t*(1 - I))/2 - I*exp(t*(1 + I))/2,                                        I*t*exp(t*(1 + I))/(2*(t - t*(1 - I))) - I*(t/(t - t*(1 - I)) + 2*I)*exp(t*(1 - I))/2]])
```
With this patch, the output is
```
Matrix([
[exp(t)*cos(t), -exp(t)*sin(t)],
[exp(t)*sin(t),  exp(t)*cos(t)]])
```

**Comment by vishalg2235** [comment]
Ping @normalhuman

**Review by normalhuman — APPROVED** [review]

**Comment by jksuom** [comment]
Looks good to me too, thanks.

**Pull request merged**